objects/spreadable.go

```diff
@@ -0,0 +1,7 @@
+package objects
+
+// Spreadable represents an object that can be spread
+type Spreadable interface {
```

Comment:
Unless we want to merge Spreadable into Object, I think we should embed `Object` interface here to force `Spreadable` is always `Object`:

```golang
type Spreadable interface {
   Object

   // Spread should return a slice of Objects
   Spread() []Object
}
```

Reply:
I merged `Spreadable` into `Object` to keep with the pattern we used for `Callable` and `Indexable`/`IndexSettable`